src/components/CPU/BreakpointPanel.js

```diff
@@ -0,0 +1,209 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import './BreakpointPanel.css';
+import { ContextMenuTrigger } from 'react-contextmenu';
+import listeners from '../../utils/listeners';
+import BreakpointContextMenu from './BreakpointPanelContextMenu';
+import BreakpointModal from './BreakpointModal';
+import BreakpointPanelItem from './BreakpointPanelItem';
+import { hasContextMenu } from '../../utils/dom';
+import { useDebuggerContext } from '../DebuggerContext';
+
+export default function BreakpointPanel(props) {
+	const context = useDebuggerContext();
+
+	const [cpuBreakpoints, setCpuBreakpoints] = useState([]);
+	const [memoryBreakpoints, setMemoryBreakpoints] = useState([]);
+	const [editingBreakpoint, setEditingBreakpoint] = useState(null);
+	const [creatingBreakpoint, setCreatingBreakpoint] = useState(false);
+	const [selectedRow, setSelectedRow] = useState(-1);
+
+	const breakpoints = [...memoryBreakpoints, ...cpuBreakpoints];
+
+	const fetchCpuBreakpoints = useCallback(() => {
+		if (!context.gameStatus.started) {
+			return;
+		}
+		context.ppsspp.send({
+			event: 'cpu.breakpoint.list',
+		}).then(({ breakpoints }) => {
+			setCpuBreakpoints(breakpoints);
+		}, () => {
+			setCpuBreakpoints([]);
+		});
+	}, [context.ppsspp, context.gameStatus.started]);
+
+	const fetchMemoryBreakpoints = useCallback(() => {
+		if (!context.gameStatus.started) {
+			return;
+		}
+		context.ppsspp.send({
+			event: 'memory.breakpoint.list',
+		}).then(({ breakpoints }) => {
+			setMemoryBreakpoints(breakpoints);
+		}, () => {
+			setMemoryBreakpoints([]);
+		});
+	}, [context.ppsspp, context.gameStatus.started]);
+
+	const fetchBreakpoints = useCallback(() => {
+		fetchCpuBreakpoints();
+		fetchMemoryBreakpoints();
+	}, [fetchCpuBreakpoints, fetchMemoryBreakpoints]);
+
+	const clearBreakpoints = useCallback(() => {
+		setCpuBreakpoints([]);
+		setMemoryBreakpoints([]);
+		setSelectedRow(-1);
+	}, []);
+
+	useEffect(() => {
+		const interval = setInterval(() => {
+			fetchBreakpoints();
+		}, 3000);

```

Comment:
Hm, is there a reason for this specific delay?

-[Unknown]

Reply:
No. I just would like to have the changes made in other debuggers _eventually_ be visible everywhere. It probably should be PPSSPP job to send events for cases like this. 
I think I'll remover this block entirely. From what I tested none of the other views (regs, dissamebly) update after making changes in Windows debugger. It's just not consistent to have pulling updates for single component.